openvpn.c

```diff
@@ -1281,6 +1281,25 @@ void OnByteCount(connection_t *c, char *msg)
             LoadLocalizedString(IDS_NFO_BYTECOUNT, in, out));
 }
 
+/*
+ * Handle INFOMSG from OpenVPN. At the moment in only handles
+ * "OPEN_URL:<url>" message used by web-based extra authentication.
+ */
+void OnInfoMsg(connection_t* c, char* msg)
+{
+    PrintDebug(L"OnInfoMsg with msg = %S", msg);
+
+    if (strstr(msg, "OPEN_URL:"))
+    {
```

Comment:
I think we should be more precise here using strbegins(). The format of <EXTRA> is not adequately documented but there should be no leading spaces matching with other message types like >INFO:, >NEED-OK and >PASSWORD: Anyway, using msg+9 below assumes this.

Looks good otherwise -- only stare-tested :)

Reply:
Agreed.